core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -949,7 +949,13 @@ class DAGScheduler(
     // serializable. If tasks are not serializable, a SparkListenerStageCompleted event
     // will be posted, which should always come after a corresponding SparkListenerStageSubmitted
     // event.
-    outputCommitCoordinator.stageStart(stage.id)
+    stage match {
+      case s: ShuffleMapStage =>
+        outputCommitCoordinator.stageStart(stage = s.id, maxPartitionId = s.numPartitions - 1)
```

Comment:
A shuffle map stage's maximum partition id is determined by the number of partitions in the RDD being computed.